Update README for Element.matches() browser support - theme a11y

tauthomas01 opened the pull request:
### What are you trying to accomplish in this PR
Fixes https://github.com/Shopify/theme-scripts/issues/83

Some of the functions inside Theme A11y require to use `matches()` method. Specifically on IE, the method will prompt an error since it is not supported officially. We need to use `msMatchesSelector` instead.

[MDN polyfill](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches#Polyfill)

#### Expectation
We should not see the error again.

![image](https://user-images.githubusercontent.com/658169/54631490-7c053300-4a52-11e9-9c6d-07c73359f800.png)

NathanPJF commented:
Thanks for bringing this up, @tauthomas01 

Instead of putting a polyfill in the script, perhaps we should follow the example of `theme-cart` where it acknowledges browser support limitations of and suggests a solution to developers: [Theme Cart - Browser Support](https://github.com/Shopify/theme-scripts/blob/6d5d5916bd82aab291780c56ddd9944f2243d6d3/packages/theme-cart/README.md#browser-support)

This same approach is being suggested in the `theme-address` update: [Theme Address - Browser Support](https://github.com/Shopify/theme-scripts/pull/78/files#diff-c55b2a4d14213a4c69e1e082b5fa0b1eR28). Product form also has a [browser support](https://github.com/Shopify/theme-scripts/blob/9f262f324bb3c4dd6f0564e678be939f7defd45c/packages/theme-product-form/README.md#browser-support) section and has commented code saying `fetch` is required.

So in the case of this PR, I'm suggesting to remove the `polyfill()` function and change the README instead to have a Browser Support section and a suggested solution.  The suggested solution could be to import a package like [`element-matches`](https://www.npmjs.com/package/element-matches) or showing the code you already wrote and telling them they can write their own.

tauthomas01 commented:
Thank you @NathanPJF for the suggestion. I like the new approach!

My new commit includes : 
- Removed the polyfill
- Added a "Browser Support" section in the README.

chrisberthe commented on packages/theme-a11y/README.md:
```diff
@@ -23,6 +23,26 @@ These files make Theme A11y accessible via the `Shopify.theme.a11y` global varia
 
+## Browser Support
+
+Theme A11y uses method not available to legacy browsers: `Element.matches()`. If you wish to support legacy browsers, make sure you add the following dependencies to your project:
```
```suggestion
Theme A11y uses a method not available in legacy browsers: `Element.matches()`. If you wish to support legacy browsers, make sure you add the following dependencies to your project:
```
